tests/storage/glusterfs/conftest.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+
+def _setup_host_with_glusterfs(host):
+    assert not host.file_exists('/usr/sbin/glusterd'), \
+        "glusterfs-server must not be installed on the host at the beginning of the tests"
+    host.yum_install(['glusterfs-server'], save_state=True)
+    host.ssh(['systemctl', 'enable', '--now', 'glusterd.service'])
+
+    for h in host.pool.hosts:
+        hostname_or_ip = h.hostname_or_ip
+        if hostname_or_ip != host.hostname_or_ip:
+            host.ssh(['iptables', '-I', 'INPUT', '-p', 'all', '-s', hostname_or_ip, '-j', 'ACCEPT'])
+
+    # Make rules reboot-persistent
+    for service in ['iptables', 'ip6tables']:
+        host.ssh(['cp', '/etc/sysconfig/%s' % service, '/etc/sysconfig/%s.orig' % service])
+        host.ssh(['service', service, 'save'])
+
+def _teardown_host_with_glusterfs(host):
+    try:
+        host.ssh(['systemctl', 'disable', '--now', 'glusterd.service'])
+    except Exception:
+        pass
+    host.yum_restore_saved_state()
+
+    for h in host.pool.hosts:
+        hostname_or_ip = h.hostname_or_ip
+        if hostname_or_ip != host.hostname_or_ip:
+            host.ssh(['iptables', '-D', 'INPUT', '-p', 'all', '-s', hostname_or_ip, '-j', 'ACCEPT'])
+
+    for service in ['iptables', 'ip6tables']:
+        host.ssh(['mv', '/etc/sysconfig/%s.orig' % service, '/etc/sysconfig/%s' % service])
+
+@pytest.fixture(scope='session')
+def pool_with_glusterfs(host):
+    for h in host.pool.hosts:
+        _setup_host_with_glusterfs(h)
+
+    yield
+    for h in host.pool.hosts:
+        _teardown_host_with_glusterfs(h)
+
+@pytest.fixture(scope='session')
+def gluster_disk(host, sr_disk):
+    print(">> Format sr_disk %s and mount it" % sr_disk)
+    device = '/dev/' + sr_disk
+    host.ssh(['mkfs.ext4', '-F', device])
+    host.ssh(['mkdir', '-p', '/mnt/sr_disk'])
+    host.ssh(['cp', '-f', '/etc/fstab', '/etc/fstab.orig'])
+    host.ssh(['echo', '/dev/sdb /mnt/sr_disk ext4 defaults 0 0', '>>/etc/fstab'])
+    host.ssh(['mount', '/mnt/sr_disk'])
+    yield
+    host.ssh(['cp', '-f', '/etc/fstab.orig', '/etc/fstab'])
+    try:
+        host.ssh(['umount', '/mnt/sr_disk'])
+        host.ssh(['rmdir', '/mnt/sr_disk'])
+    except Exception:
+        pass
+
+@pytest.fixture(scope='session')
+def gluster_volume_started(host, gluster_disk):
+    hosts = host.pool.hosts
+    assert len(hosts) > 1, "GlusterFS must be run on at least a 2 hosts pool"
+
+    host_probed = False
+    for h in hosts:
+        h.ssh(['mkdir', '-p', '/mnt/sr_disk/vol0/brick0'])
+        if h.hostname_or_ip != host.hostname_or_ip:
```

Comment:
You can avoid testing for hostname_or_ip by iterating over hosts[1:] instead since the master host is always the first one in the pool. Or, to make it more readable, create a `Pool.members` method (decorated with `@property` to be used as an attribute).